### PR TITLE
task isolation

### PR DESCRIPTION
**Pull request by xjlgod**

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -69,6 +68,10 @@ Response:
 
 #### ConnectDB  API
 
+Header:
+
+Authorization 1 cm9vdDoxMjM=
```
Comment (veezhang) [inline server/README.md]: What does it means?

File: server/README.md [hunk 2]
```diff
@@ -82,8 +85,9 @@ Response:
     "code": 0,
     "message": "",
     "data": {
-        "nsid": "e870674d-6ebc-4d9d-a1f7-bf59fdca24e8",
-        "version": "v2.6"
+        "nsid": "da821d0e-1bc5-4407-ab04-9dbcfa1dd3fb",
+        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuZWJ1bGFBZGRyZXNzIjoiMTkyLjE2OC44LjIzMzo5NjY5IiwidXNlcm5hbWUiOiJyb290In0.zP6LxNByhG1XGrzEowljBz6UkDfbm66e7Njd3zHdSio",
```
Comment (veezhang) [inline server/README.md]: Why not put it in header?

File: server/pkg/webserver/base/scheme.go [hunk 3]
```diff
@@ -75,3 +82,37 @@ func SetRoute(r router.Party, route *Route) {
...
+	for _, hook := range hooks {
+		handlers = append(handlers, hookHandler(hook, rt.Desc))
+	}
+	return handlers
+}
+
+func hookHandler(h Hook, desc string) iris.Handler {
+	return func(ctx iris.Context) {
+		// pass route description to handler
+		ctx.Values().Set("routeDesc", desc)
```
Comment (veezhang) [inline server/pkg/webserver/base/scheme.go]: What is this for?

File: server/pkg/webserver/middleware/auth.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+var (
+	mySecret     = []byte("login secret")
+	WhiteListMap = map[string]struct{}{
+		"POST/api-nebula/db/connect": {},
+	}
+)
+
+func GetLoginTokenHandler(nebulaAddress string, username string) string {
```
Comment (veezhang) [inline server/pkg/webserver/middleware/auth.go]: `GetLoginToken`?

File: server/pkg/webserver/middleware/auth.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
+package middleware
+
+import (
+	"github.com/iris-contrib/middleware/jwt"
+	"github.com/kataras/iris/v12"
+)
+
+var (
+	mySecret     = []byte("login secret")
```
Comment (veezhang) [inline server/pkg/webserver/middleware/auth.go:12]: Make it configurable?

File: server/pkg/webserver/middleware/auth.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		"POST/api-nebula/db/connect": {},
+	}
+)
+
+func GetLoginTokenHandler(nebulaAddress string, username string) string {
+	token := jwt.NewTokenWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"nebulaAddress": nebulaAddress,
+		"username":      username,
+	})
+	tokenString, _ := token.SignedString(mySecret)
```
Comment (veezhang) [inline server/pkg/webserver/middleware/auth.go]: Is it impossible to fail here?

File: server/pkg/webserver/middleware/auth.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	if _, ok := WhiteListMap[url]; ok {
+		ctx.Next()
+		return nil
+	}
+
+	j := jwt.New(jwt.Config{
+		ValidationKeyGetter: func(token *jwt.Token) (interface{}, error) {
+			return mySecret, nil
+		},
+		Expiration:    false,
```
Comment (veezhang) [inline server/pkg/webserver/middleware/auth.go]: Will it never expire? Is it possible to set an expiration time?

File: server/README.md (hunk 1 above)
Comment (xjlgod) [reply server/README.md]: the first 1 is useless，cm9vdDoxMjM= is the token

File: server/README.md (hunk 2 above)
Comment (xjlgod) [reply server/README.md]: Ok，I put them in cookie

File: server/pkg/webserver/base/scheme.go (hunk 3 above)
Comment (xjlgod) [reply server/pkg/webserver/base/scheme.go]: Pass route description to handler.

File: server/pkg/webserver/middleware/auth.go (hunk 4 above)
Comment (xjlgod) [reply server/pkg/webserver/middleware/auth.go]: Ok

File: server/pkg/webserver/middleware/auth.go (hunk 5 above)
Comment (xjlgod) [reply server/pkg/webserver/middleware/auth.go:12]: Do it in next version.

File: server/pkg/webserver/middleware/auth.go (hunk 7 above)
Comment (xjlgod) [reply server/pkg/webserver/middleware/auth.go]: I have set it.

File: server/README.md (hunk 1 above)
Comment (veezhang) [reply server/README.md]: If it's useless, why is it here?

File: server/pkg/webserver/base/scheme.go (hunk 3 above)
Comment (veezhang) [reply server/pkg/webserver/base/scheme.go]: So why need this? where to used?

File: server/pkg/webserver/middleware/auth.go (hunk 5 above)
Comment (veezhang) [reply server/pkg/webserver/middleware/auth.go:12]: Please add TODO here.

File: server/pkg/webserver/middleware/auth.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+	}
+)
+
+func GetLoginToken(nebulaAddress string, username string) (string, error) {
+	now := time.Now()
+	token := jwt.NewTokenWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"nebulaAddress": nebulaAddress,
+		"username":      username,
+		"iat":      now.Unix(),
+		"exp":      now.Add(24 * time.Hour).Unix(),
```
Comment (veezhang) [inline server/pkg/webserver/middleware/auth.go]: Can it be refreshed automatically?

File: server/pkg/webserver/base/scheme.go (hunk 3 above)
Comment (xjlgod) [reply server/pkg/webserver/base/scheme.go]: I have del it

File: server/pkg/webserver/middleware/auth.go (hunk 5 above)
Comment (xjlgod) [reply server/pkg/webserver/middleware/auth.go:12]: > Please add TODO here.
// TODO: Make it configurable
	mySecret     = []byte("login secret") 
I hava added it.

Comment (veezhang) [reply server/pkg/webserver/middleware/auth.go:12]: 👌

**Review by veezhang — APPROVED** [review]
LGTM

**Review by veezhang — APPROVED** [review]
LGTM

**Pull request merged**